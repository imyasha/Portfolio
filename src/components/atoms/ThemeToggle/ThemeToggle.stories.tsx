import { ThemeToggle } from './ThemeToggle';

export default {
	title: 'atoms/ThemeToggle',
	component: ThemeToggle,
};

export const ThemeToggleStory = () => <ThemeToggle />;

ThemeToggleStory.storyName = 'ThemeToggle';
