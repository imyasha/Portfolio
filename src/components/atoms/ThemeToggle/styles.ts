import styled from 'styled-components';

const StyledThemeToggle = styled.button`
	background: transparent;
	border: 1px solid ${({ theme }) => theme.text};
	color: ${({ theme }) => theme.text};
	border-radius: 3px;
	width: 2rem;
	height: 2rem;
	transition: 0.2s;
	cursor: pointer;

	&:hover {
		color: ${({ theme }) => theme.headline};
	}
`;

export { StyledThemeToggle };
