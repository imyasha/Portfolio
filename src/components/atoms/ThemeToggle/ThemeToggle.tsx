import { StyledThemeToggle } from './styles';

import { getTheme } from 'Redux/selectors';
import { setTheme } from 'Redux/actions/theme';
import { useDispatch, useSelector } from 'react-redux';

import Dark from 'Public/assets/dark.svg';
import Light from 'Public/assets/light.svg';

const ThemeToggle = () => {
	const theme = useSelector(getTheme);
	const dispatch = useDispatch();

	const lightTheme = theme === 'light';

	const handleToggle = () => {
		dispatch(setTheme(lightTheme ? 'dark' : 'light'));
	};

	return (
		<StyledThemeToggle
			onClick={handleToggle}
			title={`Switch to ${lightTheme ? 'dark' : 'light'} theme`}
		>
			{lightTheme ? <Dark /> : <Light />}
		</StyledThemeToggle>
	);
};

export { ThemeToggle };
