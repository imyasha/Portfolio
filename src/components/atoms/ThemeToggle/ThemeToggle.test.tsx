import { render } from '@testing-library/react';
import { ThemeToggle } from '../ThemeToggle';

const mockDispatch = jest.fn();

jest.mock('react-redux', () => ({
	useSelector: jest.fn(),
	useDispatch: () => mockDispatch,
}));

describe('ThemeToggle', () => {
	it('renders correctly', () => {
		const { container } = render(<ThemeToggle />);
		expect(container).toMatchSnapshot();
	});
});
