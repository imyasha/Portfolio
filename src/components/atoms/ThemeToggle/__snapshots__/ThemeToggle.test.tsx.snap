// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeToggle renders correctly 1`] = `
<div>
  <button
    class="styles__StyledThemeToggle-sc-1dfmkyd-0 iOJryI"
    title="Switch to light theme"
  >
    <test-file-stub />
  </button>
</div>
`;
